OpenRA.Mods.Common/UpdateRules/UpdatePath.cs

```diff
@@ -72,31 +72,31 @@ public class UpdatePath
 				new RemoveSmokeTrailWhenDamaged(),
 				new ReplaceCrateSecondsWithTicks(),
 				new UseMillisecondsForSounds(),
-				new UnhardcodeSquadManager(),
 				new RenameSupportPowerDescription(),
 				new AttackBomberFacingTolerance(),
 				new AttackFrontalFacingTolerance(),
 				new RenameCloakTypes(),
 				new SplitNukePowerMissileImage(),
 				new ReplaceSequenceEmbeddedPalette(),
-				new UnhardcodeBaseBuilderBotModule(),
 				new UnhardcodeVeteranProductionIconOverlay(),
 				new RenameContrailProperties(),
 				new RemoveDomainIndex(),
 				new AddControlGroups(),
+				new UnhardcodeSquadManager(),
```

Comment:
Add comments here and below explaining that these resolve rules so must be run last?

Reply:
Can't we automatically sort these while running the utility?

Reply:
I dunno if we want to. Some rules will give different results depending on when they are executed

Reply:
It doesn't make sense to me to run these rules before the other rules that update the yaml to not crash.

Reply:
Added, though I'm not sure if this is how they should look like though.

Reply:
> I dunno if we want to. Some rules will give different results depending on when they are executed

Add a `int` weight to each rule? Or a stage/pass at which it should run? I.e. 'pre compile'. Perhaps distinguish between rules that modify the tree structure and those that only adjust matches. 

Or have each rule maintani a string/class-name list of rules that should run prior.

Reply:
Automatic sorting would be terrible, IMO. Update rules are supposed to run in a certain order. Otherwise, they may malfunction and the update path is the place where we define the order.

Reply:
#20847 sounds like you want dependency management like we do with `Requires` for traits?

Reply:
Keep in mind that we should also be able to run individual rules. I have no idea how that would look.